scripts/shell.lua

```diff
@@ -305,6 +305,33 @@ function shell.read(replaceChar, size, history, colorFn)
         elseif k == "end" then
           strPos = #str + 1
           term.blink = 0
+        elseif k == "tab" and fileTabComplete then
+          local subStrPos = 1
+          local newStr = { }
+          for subStr in string.gmatch(str, "[^%s]+") do
+            if strPos >= subStrPos and strPos <= subStrPos + #subStr then
+              local bOk, baseDir = pcall(fs.getBaseDir, subStr)
+              if not bOk then
+                baseDir = string.sub(subStr, 1, 1) == "/" and "/" or ""
+              end
+              local file = baseDir == "" and subStr
+                        or baseDir == "/" and string.sub(subStr, 2)
+                        or string.sub(subStr, #baseDir + 2)
+              local listing = fs.list(baseDir)
```

Comment:
We can just do this, and then repeat for all directories in `shell.config.path`, although we should check that `file` doesn't contain any path separators before checking against `path` directories